legend-engine-xts-data-space/legend-engine-xt-data-space-generation/src/main/java/org/finos/legend/engine/generation/analytics/DataSpaceAnalyticsHelper.java

```diff
@@ -270,54 +275,75 @@ else if (elementDoc instanceof Root_meta_pure_metamodel_dataSpace_analytics_Data
-                                generatorExtensions.flatCollect(e -> e.getExtraExtensions(pureModel)),
-                                generatorExtensions.flatCollect(PlanGeneratorExtension::getExtraPlanTransformers)
-                        ).plan.rootExecutionNode.resultType);
-                        executableAnalysisResult.datasets = LazyIterate.flatCollect(entitlementServiceExtensions, extension -> extension.generateDatasetSpecifications(null, pureModel.getRuntimePath(execution._runtime()), execution._runtime(), HelperModelBuilder.getElementFullPath(execution._mapping(), pureModel.getExecutionSupport()), execution._mapping(), pureModelContextData, pureModel)).toList();
-                        result.executables.add(executableAnalysisResult);
+                        executableAnalysisResult.infoList.add(serviceExecutableInfo);
+                        theLambdaFunc = (LambdaFunction<?>) execution._func();
+                        theMapping = execution._mapping();
+                        theRuntime = execution._runtime();
+                    }
+                    else if (service._execution() instanceof Root_meta_legend_service_metamodel_PureMultiExecution)
+                    {
+                        Root_meta_legend_service_metamodel_PureMultiExecution execution = ((Root_meta_legend_service_metamodel_PureMultiExecution) service._execution());
+                        for (Root_meta_legend_service_metamodel_KeyedExecutionParameter keyedExecutionParameter: execution._executionParameters())
+                        {
+                            DataSpaceServiceExecutableInfo serviceExecutableInfo = new DataSpaceServiceExecutableInfo();
+                            serviceExecutableInfo.pattern = service._pattern();
+                            serviceExecutableInfo.key = keyedExecutionParameter._key();
+                            serviceExecutableInfo.query = ((PureMultiExecution) serviceProtocol.execution).func.accept(DEPRECATED_PureGrammarComposerCore.Builder.newInstance().withRenderStyle(RenderStyle.PRETTY).build());
+                            serviceExecutableInfo.mapping = HelperModelBuilder.getElementFullPath(keyedExecutionParameter._mapping(), pureModel.getExecutionSupport());
+                            serviceExecutableInfo.runtime = pureModel.getRuntimePath(keyedExecutionParameter._runtime());
+                            executableAnalysisResult.infoList.add(serviceExecutableInfo);
+                        }
+                        //for multi execution, we use the first execution to generate execution plan
+                        theLambdaFunc = (LambdaFunction<?>) execution._func();
+                        theMapping = execution._executionParameters().getFirst()._mapping();
+                        theRuntime = execution._executionParameters().getFirst()._runtime();
                     }
+                    executableAnalysisResult.result = buildExecutableResult(PlanGenerator.generateExecutionPlanDebug(
```

Comment:
nice! we have to be super sure that we only call `generatePlan` once to not punish performance.